handle external paths

johnnyaug opened the pull request:
Resolves #39.

Allow lakeFS Iceberg catalog to work with external (non-lakeFS) paths.
This way, an imported table which includes such paths, can still work.

## Assumptions
1. Used `lakectl import` to import a Iceberg table.
2. The original Iceberg table is in Hadoop Iceberg Catalog format.
3. When querying the table, the client has access to the original storage.
4. (If original table is in S3) Per-bucket configuration is used to point S3A to the lakeFS S3 gateway, but only for the specific repository name. For other bucket names, it uses AWS S3.

nopcoder reviewed (approved):
Like the idea ;)

Think we can just check `path.startsWith("s3a://")` once to branch between lakefs or non lakefs flow.

johnnyaug commented:
@nopcoder this function may be called with all of the following:
1. "Relative" lakeFS paths.
2. s3a URIs under the repository.
3. External s3a URIs.

So we need to handle all of these cases. Added a test to reflect that.
Also thanks to @lynnro314, changed the initial check to allow non-s3a external URIs.

lynnro314 reviewed (approved):
LGTM, thanks!
